types/ramda/index.d.ts

```diff
@@ -2618,6 +2741,17 @@ export function splitEvery(a: number): {
 export function splitWhen<T, U>(pred: (val: T) => boolean, list: readonly U[]): U[][];
```

Comment:
`splitWhen`'s type is wrong, Would you like to optimize it?

```ts
export function splitWhen<T>(pred: (val: T) => boolean, list: readonly T[]): T[][];
export function splitWhen<T>(pred: (val: T) => boolean): (list: readonly T[]) => T[][];
```

Reply:
ah, so that was what i missed.
I had the typing as this in the overhaul PR - should I use yours, or this?
```typescript
export function splitWhen<T>(pred: (val: T) => boolean, list: readonly T[]): [T[], T[]];
export function splitWhen<T>(pred: (val: T) => boolean): <U extends T>(list: readonly U[]) => [U[], U[]];
```

Reply:
> ah, so that was what i missed. I had the typing as this in the overhaul PR - should I use yours, or this?
> 
> ```ts
> export function splitWhen<T>(pred: (val: T) => boolean, list: readonly T[]): [T[], T[]];
> export function splitWhen<T>(pred: (val: T) => boolean): <U extends T>(list: readonly U[]) => [U[], U[]];
> ```

I think Yours are better than mine.